### PR TITLE
Enable use of refresh token (CU-8685pnbgd)

### PR DESCRIPTION
**Pull request by DudeMason**
# Description of Changes

This adds the refresh token to the tokens object retrievable from the Authenticator.

## Resolves
[CU-8685pnbgd](https://app.clickup.com/t/8685pnbgd)

## Checklist

- [x] I have labeled this PR appropriately. 
- [x] I have commented my code in hard-to-understand areas.
- [x] I have created tests to show the ticket has been resolved.
- [x] I have self-reviewed my completed code.
- [x] I have tested my changes as well as any areas that may be affected by the change.

## Additional Information

Tested on Retain

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,9 @@ $providerId = $tokensResponse->getProvider()->getIdentifier();
 // Get access_token.
 $accessToken = $tokensResponse->getAccessToken();
 
+// Get refresh_token.
+$accessToken = $tokensResponse->getRefreshToken();
```
Comment (kofipinnacle) [inline README.md]: I know it is a readme but the variable should say refreshToken

**Review by kofipinnacle — CHANGES_REQUESTED** [review]
Left you 1 comment

File: README.md (hunk 1 above)
Comment (DudeMason) [reply README.md]: Oh, good catch! Haha. Thanks!

**Review by kofipinnacle — DISMISSED** [review]
Approved

File: src/Tokens/TokensRequestor.php [hunk 2]
```diff
@@ -106,6 +107,11 @@ private static function tokensFromJsonResponse(stdClass $jsonResponse): Tokens
 
         $accessToken = new AccessToken($jsonResponse->access_token);
 
+        $refreshToken = null;
+        if (isset($jsonResponse->refresh_token)) {
```
Comment (ChrisSchwerdt) [inline src/Tokens/TokensRequestor.php]: I think we should also check if it's empty before trying to instantiate a refresh token. Something like `&& trim($jsonResponse->refresh_token) !== '')`. Can you also add a test case for the situation of an empty `refresh_token` string?

File: tests/Unit/Tokens/TokensRequestorTest.php [hunk 3]
```diff
@@ -223,6 +226,61 @@ public function fetchTokensForAuthorizationCode_ResponseMissingIdToken_ThrowsExp
...
+        $tokensRequestor = new TokensRequestor($provider, $redirectUri, $challenge);
+
+        // Act
+        $tokens = $tokensRequestor->fetchTokensForAuthorizationCode(new AuthorizationCode('authorization-code'));
+
+        // Assert
+        $this->assertInstanceOf(Tokens::class, $tokens);
+        $this->assertEquals($tokens->getUserIdToken(), new UserIdToken($userIdToken));
+        $this->assertSame($tokens->getAccessToken()->getValue(), $accessToken);
+        $this->assertSame($tokens->getRefreshToken(), null);
```
Comment (ChrisSchwerdt) [inline tests/Unit/Tokens/TokensRequestorTest.php]: Might be clearer to just `assertNull()`.

**Review by ChrisSchwerdt — CHANGES_REQUESTED** [review]

File: src/Tokens/TokensRequestor.php (hunk 2 above)
Comment (DudeMason) [reply src/Tokens/TokensRequestor.php]: Good point, I'll do that!

**Review by ChrisSchwerdt — APPROVED** [review]
